fragment.go

```diff
@@ -0,0 +1,122 @@
+package vite
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+)
+
+// HTMLFragment generates an HTML fragment for Vite integration based on the provided configuration.
+//
+// This function takes a Config struct and uses it to create the necessary HTML
+// elements for including Vite-managed assets in a web page. The returned HTML
+// is typically intended to be placed in the <head> section of an HTML document.
+//
+// Parameters:
+//   - config: A Config struct containing the necessary configuration options for Vite integration.
+//
+// Returns:
+//   - template.HTML: An HTML fragment as a template.HTML type, which can be safely embedded in HTML templates.
+//   - error: An error if the HTML fragment generation fails for any reason.
+//
+// The returned HTML fragment may include elements such as:
+//   - Script tags for Vite client in development mode
+//   - Links to stylesheets
+//   - Module preload tags
+//   - Script tags for entry points and other necessary JavaScript
+//
+// Usage example:
+//
+//	fragment, err := HTMLFragment(myConfig)
+//	if err != nil {
+//	    // Handle error
+//	}
+//	// Use fragment in your HTML template
+func HTMLFragment(config Config) (template.HTML, error) {
+	pd := &pageData{
+		IsDev:     config.IsDev,
+		ViteEntry: config.ViteEntry,
+		ViteURL:   config.ViteURL,
+	}
+
+	if config.IsDev {
+		pd.PluginReactPreamble = template.HTML(PluginReactPreamble(config.ViteURL))
+	} else {
+		mf, err := config.FS.Open(".vite/manifest.json")
```

Comment:
@ge3224 Do we need a config option for the path to the manifest file?

Reply:
I see in [Vite's docs](https://vitejs.dev/config/build-options.html#build-manifest) that the file name can be configured. I'm not sure what that looks like at the moment. We'll have to test and consider how to implement. Because Vite's config accepts a union type, `boolean | string`, I'm thinking we would handle this in Go with perhaps a struct?

```go
type Config struct {
  //...
  Manifest struct {
    Enabled bool,
    Filename string,
  }
  //...
}
```

Reply:
I don't think we need to implement it. That [`build.manifest` setting in Vite](https://vitejs.dev/config/build-options.html#build-manifest) is just a configuration option that a) decides whether to generate the manifest at all and b) where the generated file will be written to. So if `false`, there will be no manifest, if `true` there will be a manifest in `.vite/manifest.json`, and if it's a string, the manifest will be written to that file.

We only need the manifest file though, regardless how it's named.

What we could accept is a path to the file name of the manifest, and use `.vite/manifest.json` by default.

Reply:
So I think it's more of a new configuration options `ViteManifest string` that has a default of `.vite/manifest.json`, and then have the code like this ...

```go
	if config.IsDev {
		pd.PluginReactPreamble = template.HTML(PluginReactPreamble(config.ViteURL))
	} else {
		mf, err := config.FS.Open(config.ViteManifest)
```

Reply:
I've added the `ViteManifest` option in 6e0568e06042be365058ceb81111e9381ea62786. Make sure to refetch.